### PR TITLE
#619 - wait for hydra-maester deployment to be running

### PR DESCRIPTION
**Pull request by veichtj**
In case hydra-maester deployment is rolled out to trigger OAuth2Client synchronization, reconciler needs to wait till deployment is up. Otherwise it might happen that following tests not running successfully as the clients are not yet available

### CODE REVIEW

File: pkg/reconciler/instances/ory/hydra/hydra.go [hunk 1]
```diff
@@ -2,12 +2,12 @@ package hydra
```
Comment (raypinto) [inline pkg/reconciler/instances/ory/hydra/hydra.go:2]: How about renaming this file and folder to syncer or hydraSyncer?

File: pkg/reconciler/instances/ory/k8s/rollout.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+//go:generate mockery --name=RolloutHandler --outpkg=mock --case=underscore
+//RolloutHandler exposes functionality to rollout k8s objects
+type RolloutHandler interface {
+	//Rollout a given deployment and wait till it successfully up
+	RolloutAndWaitForDeployment(ctx context.Context, deployment, namespace string, client internalKubernetes.Client, logger *zap.SugaredLogger) error
+}
+
+type DefaultRolloutHandler struct{}
+
+func NewDefaultRolloutHandler() *DefaultRolloutHandler {
```
[the remark this answers is not in the document]
Comment (veichtj) [reply pkg/reconciler/instances/ory/k8s/rollout.go:28]: Well sure a comment could be added, but the `NewDefaultRolloutHandler ` is just a constructor without any additional functionality, so the comment would basically repeat that what the name says

File: pkg/reconciler/instances/ory/hydra/hydra.go (hunk 1 above)
Comment (veichtj) [reply pkg/reconciler/instances/ory/hydra/hydra.go:2]: Yes, good proposal, I've renamed it

**Review by raypinto — APPROVED** [review]

**Review by strekm — APPROVED** [review]
/meow

**Review by suleymanakbas91 — APPROVED** [review]

**Pull request merged**